File: frontend/src/models/Landmark.ts
import { LandmarkLink } from "./LandmarkLink";
import { LandmarkMetadata } from "./LandmarkMetadata";

export interface Landmark {
  metadata: LandmarkMetadata;
  farms: string[];
  tags: string[];
  biomes: string[];
  dimension: string;
  links: LandmarkLink[];
}

File: core/src/landmarks/landmark.rs
use crate::{
    minecraft::{Biome, Dimension, Farm},
    Tag,
};
use serde::{Deserialize, Serialize};

use super::{LandmarkLink, LandmarkMetadata};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct Landmark {
    pub metadata: LandmarkMetadata,
    #[serde(default)]
    pub farms: Vec<Farm>,
    #[serde(default)]
    pub tags: Vec<Tag>,
    #[serde(default)]
    pub biomes: Vec<Biome>,
    pub dimension: Dimension,
    #[serde(default)]
    pub links: Vec<LandmarkLink>,
}
